Core/GUI/GameTooltipHooks.lua

```diff
@@ -515,14 +515,20 @@ local function onTooltipSetItem(tooltip, tooltipData)
 		return
 	end
 
-	local _, itemLink = tooltip:GetItem()
-	if type(itemLink) ~= "string" then
+	local name, link = tooltip:GetItem()
+	if type(link) ~= "string" or not name then
 		return
 	end
 
-	local id = itemLink:match("item:(%d+):")
+	local id = link:match("item:(%d*)")
 	assert(id, "Failed to extract item ID from item link (format might have changed?)")
 	processItem(tonumber(id), tooltip)
 end
 
-_G.TooltipDataProcessor.AddTooltipPostCall(_G.Enum.TooltipDataType.Item, onTooltipSetItem)
+local function onTooltipSetItemFunction(tooltip, tooltipData)
+	if tooltip == _G.GameTooltip or tooltip == _G.ItemRefTooltip then
```

Comment:
I assume this is intended to avoid the case where it's a `keystone` link? If so, isn't that eliminating the need to change the actual tooltip function, since the only type of links it would see are `item` links?

Reply:
This function reminded me of another that the game has for the PTR_IssueReporter. So, I just adapted it to Rarity, here is the function I am talking about.

```Lua
function PTR_IssueReporter.SetupItemTooltips()
    local function attachItemTooltip(self)
        local name, link = self:GetItem()
        if (link) and (name) then
            local id = string.match(link, "item:(%d*)")
			-- Professions refactor requires an update to this code in both fetching the current recipe ID and retrieving the
			-- reagent information from the mouse position.
           --if (id == "" or id == "0") and TradeSkillFrame ~= nil and TradeSkillFrame:IsVisible() and GetMouseFocus().reagentIndex then
           --    local selectedRecipe = TradeSkillFrame.RecipeList:GetSelectedRecipeID()
           --    for i = 1, 8 do
           --        if GetMouseFocus().reagentIndex == i then
           --            id = C_TradeSkillUI.GetRecipeReagentItemLink(selectedRecipe, i):match("item:(%d+):") or nil
           --            break
           --        end
           --    end
           --end
            if (id) then
                PTR_IssueReporter.HookIntoTooltip(self, PTR_IssueReporter.TooltipTypes.item, id, name, nil, nil, link)
            end
        end
    end

	local function onTooltipSetItemFunction(tooltip, tooltipData)
		if (tooltip == GameTooltip or tooltip == ItemRefTooltip) then
			attachItemTooltip(tooltip)
		end
	end
	TooltipDataProcessor.AddTooltipPostCall(Enum.TooltipDataType.Item, onTooltipSetItemFunction)
end
```